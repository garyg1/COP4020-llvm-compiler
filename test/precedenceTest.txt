extern int print_float(float x);

def int main() (
    print_float(1.0 * 2.0 / 3.0):
    print_float(1.0 + 2.0 * 3.0):
    print_float(2.0 * 3.0 + 1.0):
    
    # ':' should be low precedence
    print_float(
        1.0 * 2.0 + 3.0 - 4.0 / 5.0:
        (1.0 * 2.0 + 3.0 - 4.0) / 5.0
    )
);