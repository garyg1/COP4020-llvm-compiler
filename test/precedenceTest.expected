0.666667
7.000000
7.000000
0.200000
